### ISSUE TITLE
'Storage.connect is not available on concrete implemetations'

### ISSUE THREAD

**Issue by nathanxmeyer** [frictionlessdata/tableschema-py#278]
Hello, python novice here, experimenting with SQL Alchemy and tableschema and having trouble moving past and exception from Storage.py

Have this code:

from tableschema import Table
from sqlalchemy import create_engine
import json
from pathlib import Path

def builddb(maindir,typedir,schemadir):

     dbname = maindir / 'data/mydb.sqlite'
     s_engine = 'sqlite:////.' + str(dbname)
     engine = create_engine(s_engine)

     schemalist = schemadir.glob("*.json")
     for schema in schemalist:
         schemadata = json.load(open(schema))
         csvfile = schemadata["csv"]
         csvfile = typedir / csvfile

         try:
            
             mytable = Table(csvfile, schema=schemadata)
             tablename = schemadata["tablename"]
             mytable.save(tablename, storage='sql', engine=engine)
    
         except Exception as error:
             print(error)

It always raises the error: "Storage.connect is not available on concrete implemetations"

In Storage.py, there is this code:
...
if cls is not Storage:
     message = 'Storage.connect is not available on concrete implemetations'
     raise exceptions.StorageError(message)

Out of curiosity I commented out this IF statement any my database builds as expected. Am I doing something wrong in my implementation??

I appreciate the response and feel free to point me to documentation that explains my mistake.  Thank you.

---

Please preserve this line to notify @roll (lead of this repository)

**Comment by roll** [comment]
Hi @nathanxmeyer,

I can't reproduce the error. For example, this one works fine for me:

```
import sqlalchemy
from tableschema import Table

engine = sqlalchemy.create_engine('sqlite://')
table = Table('data/data.csv')
table.infer()
table.save('test', storage='sql', engine=engine)
print(list(engine.execute('select * from test')))
```

Can you please extract and share some minimal code/data example to reproduce the problem?

**Comment by nathanxmeyer** [comment]
Hi @roll,

Since I posted the question, I had switched over to working on GeoAlchemy2 and in the process I have updated my environment for tableschema, SQLite (including adding Spatialite) and SQLAlchemy and now I cannot reproduce it either.  I was using tableschema  1.15.3 when I encountered the error but unsure of other version numbers.  In any case it seems not  to be a problem anymore although I wish I understood why it was and now it is not.

Thank you for the response.
